test/DynamoCoreWpfTests/ViewExtensions/GraphNodeManagerViewExtensionTests.cs

```diff
@@ -205,19 +205,24 @@ public void ContainsEmptyListOrNullTest()
 
             OpenAndRun(@"pkgs\Dynamo Samples\extra\GraphNodeManagerTestGraph_NullsEmptyLists.dyn");
 
-            Utility.DispatcherUtil.DoEvents();
+            var hwm = this.ViewModel.CurrentSpace;
 
-            var view = viewExt.ManagerView;
+            int nullNodesCount = hwm.Nodes.Count(ContainsAnyNulls);
+            int emptyListNodesCount = hwm.Nodes.Count(ContainsAnyEmptyLists);
 
+            var view = viewExt.ManagerView;
             var images = WpfUtilities.ChildrenOfType<Image>(view.NodesInfoDataGrid);
 
-            int nullNodesImageCount = GetImageCount(images, "Null");
-            int emptyListNodesImageCount = GetImageCount(images, "EmptyList");
+            Utility.DispatcherUtil.DoEventsLoop(() =>
+            {
+                int nullNodesImageCount = GetImageCount(images, "Null");
+                int emptyListNodesImageCount = GetImageCount(images, "EmptyList");
 
-            var hwm = this.ViewModel.CurrentSpace;
+                return (nullNodesImageCount == nullNodesCount) && (emptyListNodesImageCount == emptyListNodesCount);
+            });
```

Comment:
DoEventsLoop will fail after 20 seconds if the function argument does not return true.